nsxt/resource_nsxt_policy_ip_address_allocation.go

```diff
@@ -125,6 +125,37 @@ func resourceNsxtPolicyIPAddressAllocationCreate(d *schema.ResourceData, m inter
 		return handleCreateError("IPAddressAllocation", id, err)
 	}
 
+	if allocationIP == "" {
+		// Wait for actual IP allocation
+		log.Printf("[DEBUG] Waiting for realization of IP Address for IP Allocation with ID %s", id)
+
+		obj, err := client.Get(poolID, id)
+		if err != nil {
+			return err
+		}
+
+		timeout := d.Get("timeout").(int)
+		stateConf := nsxtPolicyWaitForRealizationStateConf(connector, d, *obj.Path, timeout)
+		entity, err := stateConf.WaitForState()
+		if err != nil {
+			// Clean up NSX allocation
+			resourceNsxtPolicyIPAddressAllocationDelete(d, m)
+			return err
+		}
+		realizedResource := entity.(model.GenericPolicyRealizedResource)
+		for _, attr := range realizedResource.ExtendedAttributes {
+			if *attr.Key == "allocation_ip" {
+				d.SetId(id)
+				d.Set("nsx_id", id)
+				d.Set("allocation_ip", attr.Values[0])
+				return resourceNsxtPolicyIPAddressAllocationRead(d, m)
+			}
+		}
+		// Clean up NSX allocation
+		resourceNsxtPolicyIPAddressAllocationDelete(d, m)
```

Comment:
Is this "just for safety"? I don't think there is a case where realisation would succeed but no IP was allocated.

Reply:
Correct, this code should not be reachable